src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -0,0 +1,223 @@
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.ClimberConstants;
+import frc.robot.Constants.MotorIDConstants;
+import frc.robot.Robot;
+import java.util.function.BooleanSupplier;
+
+public class Climber extends SubsystemBase {
+  /** Creates a new Climber. */
+  private TalonFX climberMotorFront;
+
+  private TalonFX climberMotorBack;
+  private MotorOutputConfigs climberOutputConfigs;
+  private TalonFXConfiguration frontConfigs;
+  private TalonFXConfiguration backConfigs;
+  private DCMotor simClimberGearbox;
+  private SingleJointedArmSim climberSimNormal;
+  private SingleJointedArmSim climberSimLifting;
+  private Mechanism2d climbMech;
+  private MechanismRoot2d climbMechRoot;
+  private MechanismLigament2d climbMechLigament;
+  private MechanismLigament2d climbMechTargetLigament;
+  private Slot0Configs climberConfigsToClimber;
+  private Slot1Configs climberConfigsAtClimber;
+  private Angle climberTargetAngle;
+  private boolean isClimbing;
+
+  public Climber() {
+    climberMotorFront = new TalonFX(MotorIDConstants.kCLimberMotorLeader);
+    climberMotorBack = new TalonFX(MotorIDConstants.kCLimberMotorFollower);
+    climberConfigsToClimber =
+        new Slot0Configs()
+            .withKP(ClimberConstants.kClimberP0)
+            .withKI(ClimberConstants.kClimberI0)
+            .withKD(ClimberConstants.kClimberD0)
+            .withKG(ClimberConstants.kClimberkG0)
+            .withKV(ClimberConstants.kClimberkV0);
+    climberConfigsAtClimber =
+        new Slot1Configs()
+            .withKP(ClimberConstants.kClimberP1)
+            .withKI(ClimberConstants.kClimberI1)
+            .withKD(ClimberConstants.kClimberD1)
+            .withKG(ClimberConstants.kClimberkG1)
+            .withKV(ClimberConstants.kClimberkV1);
+    isClimbing = false;
+    // Set the configs
+    climberOutputConfigs = new MotorOutputConfigs().withNeutralMode(NeutralModeValue.Brake);
+    frontConfigs =
+        new TalonFXConfiguration()
+            .withSlot0(climberConfigsToClimber)
+            .withSlot1(climberConfigsAtClimber)
+            .withMotorOutput(
+                climberOutputConfigs.withInverted(InvertedValue.CounterClockwise_Positive));
+    backConfigs =
+        new TalonFXConfiguration()
+            .withSlot0(climberConfigsToClimber)
+            .withSlot1(climberConfigsAtClimber)
+            .withMotorOutput(climberOutputConfigs.withInverted(InvertedValue.Clockwise_Positive));
+    climberMotorFront.getConfigurator().apply(frontConfigs);
+    climberMotorBack.getConfigurator().apply(backConfigs);
+    // For simulation
+    climberTargetAngle = ClimberConstants.kClimberRetractedSetpoint;
+    if (Robot.isSimulation()) {
+      simClimberGearbox = DCMotor.getKrakenX60(2);
```

Comment:
After looking at the CAD, this should actually be only 1 Kraken, also this should be a constant